tests/shell/main.rs

```diff
@@ -1,18 +1,40 @@
+use assert_cmd::prelude::*;
 use lorri::{
     builder,
     cas::ContentAddressable,
+    cli::ShellOptions,
     ops::shell,
     project::{roots::Roots, Project},
     NixFile,
 };
 use std::fs;
 use std::iter::FromIterator;
 use std::path::{Path, PathBuf};
+use std::process::Command;
 
 #[test]
 fn loads_env() {
     let tempdir = tempfile::tempdir().expect("tempfile::tempdir() failed us!");
     let project = project("loads_env", tempdir.path());
+    let opts = ShellOptions {
+        cached: true,
+        nix_file: PathBuf::from(&project.nix_file),
+    };
+    // Try to launch it as a real user.
+    Command::cargo_bin("lorri")
+        .unwrap()
+        .args(&[
+            "shell",
+            "--shell-file",
+            PathBuf::from(&project.nix_file)
+                .as_os_str()
+                .to_str()
+                .unwrap(),
+        ])
+        .current_dir(&tempdir)
+        .assert()
+        .success();
```

Comment:
I’m not convinced that we need to include 8 new transitive dependencies for this one line. Let’s just do it manually.

Reply:
As it's in the dev-dependencies, is it really an issue ? It's only included for the test binary and not in the release binary. And I plan to add more tests about CLI so use this at different places.

Reply:
The issue here is that every transitive dependency blows up CI time, and we run these tests on CI. Every dependency is a potential point of problems in the future, and has to carry its (transitive) weight.

> And I plan to add more tests about CLI so use this at different places.

Is it an undue amount of code to just use `Command` and pattern-match on the result like we do in the other tests?

Reply:
Ok 🙂 I updated the test